functions/src/modules/SE.functions.ts

```diff
@@ -88,3 +93,230 @@ export const processNewAllotment = functions.database
+                languages: helpers.validateCellValue(row, columnsIndex, 'Languages'),
+                soundQualityRating: helpers.validateCellValue(row, columnsIndex, 'SoundQuality'),
+                soundIssues: helpers.validateCellValue(row, columnsIndex, 'SoundIssues'),
+                comments: helpers.validateCellValue(row, columnsIndex, 'Comments'),
+                submissionTimestamp: helpers.validateCellValue(row, columnsIndex, 'Timestamp'),
+                submissionSerial: helpers.validateCellValue(row, columnsIndex, 'SubmissionSerial'),
+            };
+
+
+            if (lastFile.file_name === AudioFileName) {
+                if (!skip)
+                    lastFile.chunks.push({
+                        audioFileName: AudioFileName,
+                        beginning: moment.duration(row[columnsIndex['Beginning']]).asSeconds(),
+                        ending: moment.duration(row[columnsIndex['Ending']]).asSeconds(),
+                        continuationFrom,
+                        contentReporting,
+                        importTimestamp: admin.database.ServerValue.TIMESTAMP,
+                        processingResolution: row[columnsIndex['Resolution']],
+                    });
+            } else { // NEW FILE -- Save the previous Chunks and CLEAR
+                if (lastFile.file_name != null && !lastFile.skip) {
+                    summary.addedChunks++;
+
+                    //////////////////////////////////////////
+                    //
+                    // Ensuring chunks do NOT overlap in time
+                    //
+                    //////////////////////////////////////////
+                    let lastEndingTime = null;
+                    for (let k = 0; k < lastFile.chunks.length; k++) {
+                        let chunk = lastFile.chunks[k];
+                        
+                        let beginning = moment(chunk.beginning, "HH:mm:ss");
+                        let trackNameRegex = /\w+-\d+(.*)/;            
+                        let track = trackNameRegex.exec(AudioFileName)[1];
+
+                        // check if last chunck ending time is GREATER than current chunk ending time 
+                        if (lastEndingTime && lastFile.track == track && lastEndingTime.diff(beginning)._data.seconds >= 0 ) {
+                                summary.overlappingChunks.push(lastFile);
+                                console.log('Overlapping chunks');
+                        }
+                        lastEndingTime = moment(chunk.ending, "HH:mm:ss");
+                    }
+
+
+                    ///////////////////////////////
+                    // Sorting by Beginning time
+                    ///////////////////////////////
+                    let chunks = lastFile.chunks;
+
+                    // `start` attribute is added to be used in sorting
+                    chunks.forEach(element => element.start = moment.duration(element.beginning).asSeconds());
```

Comment:
Why can't you sort by `chunk.beginning`? I think it's identical to your artificial `start` attribute.

Reply:
I though so. The question is actually just to bring your attention to this issue.